iceoryx_posh/test/moduletests/test_popo_typed_subscriber.cpp

```diff
@@ -0,0 +1,170 @@
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, getUid).Times(1);
+    // ===== Test ===== //
+    sut.getUid();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, GetsServiceDescriptionViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, getServiceDescription).Times(1);
+    // ===== Test ===== //
+    sut.getServiceDescription();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, GetsSubscriptionStateViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, getSubscriptionState).Times(1);
+    // ===== Test ===== //
+    sut.getSubscriptionState();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, SubscribesViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, subscribe).Times(1);
+    // ===== Test ===== //
+    sut.subscribe(1);
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, UnsubscribesViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, unsubscribe).Times(1);
+    // ===== Test ===== //
+    sut.unsubscribe();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, ChecksForNewSamplesViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, hasNewSamples).Times(1);
+    // ===== Test ===== //
+    sut.hasNewSamples();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, ChecksForMissedSamplesViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, hasMissedSamples).Times(1);
+    // ===== Test ===== //
+    sut.hasMissedSamples();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, ReceivesSamplesViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, take).Times(1).WillOnce(
+        Return(ByMove(iox::cxx::success<iox::cxx::optional<iox::popo::Sample<const DummyData>>>())));
+    // ===== Test ===== //
+    sut.take();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, ReleasesQueuedSamplesViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    EXPECT_CALL(sut, releaseQueuedSamples).Times(1);
+    // ===== Test ===== //
+    sut.releaseQueuedSamples();
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedSubscriberTest, SetsConditionVariableViaBaseSubscriber)
+{
+    // ===== Setup ===== //
+    auto conditionVariable = new iox::popo::ConditionVariableData();
+    // EXPECT_CALL(sut, setConditionVariable).Times(1);
+    // // ===== Test ===== //
+    // sut.setConditionVariable(conditionVariable);
+    // ===== Verify ===== //
+    // ===== Cleanup ===== //
+    delete conditionVariable;
+}
+
+TEST_F(TypedSubscriberTest, UnsetsConditionVariableViaBaseSubscriber)
+{
+    // ===== Setup ===== //
```

Comment:
The following tests do nothing. remove or implement

Reply:
fixed